src/utils/index.ts

```diff
@@ -27,22 +27,28 @@ function filterArgs(
 // Add all required search params with user inputs as values
 export function formatUrl(
   url: string,
-  params?: string[],
-  args?: Inputs,
+  paramKeys?: string[],
+  paramInputs?: Inputs,
   slug?: Inputs,
-  optionalParams?: Inputs,
+  optionalParamKeys?: string[],
+  optionalParamInputs?: Inputs,
 ) {
   const newUrl =
     slug && Object.keys(slug).length > 0
       ? new URL(Object.values(slug)[0], url) // If there's a user inputted param for the URL slug, replace the default existing slug or include it
       : new URL(url);
 
-  if (params && args) {
-    params.forEach((param: string) => {
-      if (args[param]) newUrl.searchParams.set(param, args[param]);
-      else if (optionalParams?.[param]) {
-        newUrl.searchParams.set(param, optionalParams?.[param]);
-      }
+  if (paramKeys && paramInputs) {
+    paramKeys.forEach((param: string) => {
+      if (paramInputs[param])
+        newUrl.searchParams.set(param, paramInputs[param]);
+    });
+  }
+
+  if (optionalParamKeys && optionalParamInputs) {
+    optionalParamKeys.forEach((param: string) => {
+      if (optionalParamInputs[param])
+        newUrl.searchParams.set(param, optionalParamInputs[param]);
     });
   }
```

Comment:
I think this is not working as expected? We also need to consider the default values for optional params here. In other words, I think the function needs to receive an `Inputs` object instead of `string[]` for `optionalParamKeys` (which is effectively "optional param keys with their defaults").